src/checkout.c

```diff
@@ -2521,7 +2521,8 @@ int git_checkout_iterator(
 
 	if (data.opts.baseline_index) {
 		if ((error = git_iterator_for_index(
-				&baseline, data.opts.baseline_index, &baseline_opts)) < 0)
+				&baseline, git_index_owner(data.opts.baseline_index),
```

Comment:
I checked all calls to  `git_checkout_iterator`, and `git_index_owner(data.opts.baseline_index)` will be non-null in all cases, but we could also change `git_checkout_iterator` to take an additional `repo` param.